atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java

```diff
@@ -134,80 +134,107 @@ static void validatePartitioner(Cassandra.Client client, CassandraKeyValueServic
+                if (attemptedToCreateKeyspaceTwice(ire)) {
                     // if we got this far, we're done, no need to continue on other hosts
                     someHostWasAbleToCreateTheKeyspace = true;
                     break;
-                } catch (InvalidRequestException ire) {
-                    if (attemptedToCreateKeyspaceTwice(ire)) {
-                        log.info("Attempted to create keyspace {} on multiple hosts at once", config.keyspace());
-
-                        // if we got this far, we're done, no need to continue on other hosts
-                        someHostWasAbleToCreateTheKeyspace = true;
-                        break;
-                    } else {
-                        throw ire;
-                    }
-                } catch (Exception f) {
-                    log.warn("Couldn't use host {} to create keyspace."
-                            + " It returned exception \"{}\" during the attempt."
-                            + " We will retry on other nodes, so this shouldn't be a problem unless all nodes failed."
-                            + " See the debug-level log for the stack trace.", host, f.toString(), f);
-                    log.debug("Specifically, creating the keyspace failed with the following stack trace", f);
+                } else {
+                    throw ire;
                 }
+            } catch (Exception f) {
+                log.warn("Couldn't use host {} to create keyspace."
+                        + " It returned exception \"{}\" during the attempt."
+                        + " We will retry on other nodes, so this shouldn't be a problem unless all nodes failed."
+                        + " See the debug-level log for the stack trace.", host, f.toString(), f);
+                log.debug("Specifically, creating the keyspace failed with the following stack trace", f);
             }
-            if (!someHostWasAbleToCreateTheKeyspace) {
-                throw new TException("No host tried was able to create the keyspace requested.");
-            }
+        }
+        if (!someHostWasAbleToCreateTheKeyspace) {
+            throw new TException("No host tried was able to create the keyspace requested.");
         }
     }
 
+    private static void updateExistingKeyspace(CassandraClientPool clientPool, CassandraKeyValueServiceConfig config)
+            throws TException {
+        clientPool.runWithRetry(new FunctionCheckedException<Cassandra.Client, Void, TException>() {
+            @Override
+            public Void apply(Cassandra.Client client) throws TException {
+                KsDef originalKsDef = client.describe_keyspace(config.keyspace());
+                // there was an existing keyspace
+                // check and make sure it's definition is up to date with our config
+                KsDef modifiedKsDef = originalKsDef.deepCopy();
+                CassandraVerifier.checkAndSetReplicationFactor(
```

Comment:
Mmm, smells like years-past of copy-pasting things around. I'm surprised there isn't a baseline check for "unnecessarily verbose intra-class method calls" or something